components/ingest-service/server/chef.go

```diff
@@ -101,7 +101,7 @@ func (s *ChefIngestServer) ProcessChefAction(ctx context.Context, action *chef.A
 		defer close(errc)
 
 		var err error
-		if errQ := s.chefActionPipeline.Run(ctx, action, errc); err != nil {
+		if errQ := s.chefActionPipeline.Run(ctx, action, errc); errQ != nil {
```

Comment:
This was a deadlock/memory leak. Goroutines would sit around forever if the chef action queue filled up

Reply:
💭 Could this be most of the problem you've been chasing for the customer?

Nice find.

Reply:
i did not see this in their core dump, but was able to consistently hit it on my machine when I did
`chef_load_nodes 2000`